sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -130,6 +172,156 @@ public HttpPipeline getHttpPipeline() {
+     * @param clazz The model class to convert the response to.
+     * @param <T> The generic type to convert the response to.
+     * @return The application/json digital twin
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<T> getDigitalTwin(String digitalTwinId, Class<T> clazz)
+    {
+        return getDigitalTwinWithResponse(digitalTwinId, clazz)
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Gets a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+     * @param clazz The model class to convert the response to.
+     * @param <T> The generic type to convert the response to.
+     * @return A Http response containing the application/json digital twin
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<DigitalTwinsResponse<T>> getDigitalTwinWithResponse(String digitalTwinId, Class<T> clazz)
+    {
+        return withContext(context -> getDigitalTwinWithResponse(digitalTwinId, clazz, context));
+    }
+
+    <T> Mono<DigitalTwinsResponse<T>> getDigitalTwinWithResponse(String digitalTwinId, Class<T> clazz, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .getByIdWithResponseAsync(digitalTwinId, context)
+            .map(response -> {
+                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+                DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                return new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders);
+            });
+    }
+
+    /**
+     * Updates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin
+     * @return an empty Mono
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> updateDigitalTwin(String digitalTwinId, List<Object> digitalTwinUpdateOperations)
+    {
+        return updateDigitalTwinWithResponse(digitalTwinId, digitalTwinUpdateOperations, new RequestOptions())
+            .flatMap(voidResponse -> Mono.empty());
+    }
+
+    /**
+     * Updates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin
+     * @param options The optional settings for this request
+     * @return A Http response
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<Void>> updateDigitalTwinWithResponse(String digitalTwinId, List<Object> digitalTwinUpdateOperations, RequestOptions options)
```

Comment:
Don't use the RequestOptions type directly. Each set of client side APIs should have their own request options implementation like "UpdateDigitalTwinRequestOptions" instead. That way we can add options to this API later without worrying about adding the same options to other APIs that share the same options type

Reply:
I would start a thread with Srikanata to get some guidance here as we discussed

Reply:
Feel free to start that thread then. My understanding is that they already document this style of optional parameter grouping [here](https://azure.github.io/azure-sdk/java_design.html#service-method-parameters)

Reply:
I did chat with Abhipsa later and have better understanding. I'm working on adding optional param classes for update and delete.